### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Jiadalee** [MouseLand/cellpose#1220]
**Describe the bug**

In the train.py file, I have quick questions for the loss calculation function. 

For the first loss function,it is converting the labels to .long() type. If my max labels' value is float64 ≈ 0.00392. This results in all-zero labels → trivial classification, possibly misleadingly low or zero loss.

For the second loss function in cuda gpu, if my labels has max value, let's say 0.00392  < 0.5, then all mask becomes all 0s, then possibly missleadingly zero loss. In other words, My mask lbl[:, -3] has values all ≈ 0.00392 → all < 0.5 → binary mask becomes all 0s. The loss becomes 0, because it’s comparing 0s with 0s.

Could you clarify a bit  for these conversion logics? I didn't dive to these codes in depth. Any clarification would be appreciated.

Below codes:
```
def _loss_fn_class(lbl, y, class_weights=None):
    """
    Calculates the loss function between true labels lbl and prediction y.

    Args:
        lbl (numpy.ndarray): True labels (cellprob, flowsY, flowsX).
        y (torch.Tensor): Predicted values (flowsY, flowsX, cellprob).
        
    Returns:
        torch.Tensor: Loss value.

    """

    criterion3 = nn.CrossEntropyLoss(reduction="mean", weight=class_weights)
    loss3 = criterion3(y[:, :-3], lbl[:, 0].long())
    
    return loss3

def _loss_fn_seg(lbl, y, device):
    """
    Calculates the loss function between true labels lbl and prediction y.

    Args:
        lbl (numpy.ndarray): True labels (cellprob, flowsY, flowsX).
        y (torch.Tensor): Predicted values (flowsY, flowsX, cellprob).
        device (torch.device): Device on which the tensors are located.

    Returns:
        torch.Tensor: Loss value.

    """
    criterion = nn.MSELoss(reduction="mean")
    criterion2 = nn.BCEWithLogitsLoss(reduction="mean")
    veci = 5. * lbl[:, -2:]
    loss = criterion(y[:, -3:-1], veci)
    loss /= 2.
    loss2 = criterion2(y[:, -1], (lbl[:, -3] > 0.5).float())
    loss = loss + loss2
    return loss
```
